homeassistant/components/remootio/config_flow.py

```diff
@@ -0,0 +1,187 @@
+        ): vol.All(
+            vol.Coerce(str),
+            vol.In([DEVICE_CLASS_GARAGE, DEVICE_CLASS_GATE]),
+            msg="Controlled device's class appears to be invalid",
+        ),
+    },
+    extra=REMOVE_EXTRA,
+)
+
+DEVICE_NAME = "Remootio Device"
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input."""
+
+    _LOGGER.debug("Validating input... Input [%s]", data)
+
+    INPUT_VALIDATION_SCHEMA(data)
+
+    connection_options: ConnectionOptions = ConnectionOptions(
+        data[CONF_IP_ADDRESS], data[CONF_API_SECRET_KEY], data[CONF_API_AUTH_KEY]
+    )
+
+    device_serial_number: str = await get_serial_number(
+        hass, connection_options, _LOGGER
+    )
+
+    return {
+        CONF_TITLE: f"{DEVICE_NAME} (IP: {data[CONF_IP_ADDRESS]}, S/N: {device_serial_number})",
+        CONF_SERIAL_NUMBER: device_serial_number,
+    }
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Remootio."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the class."""
+        super().__init__()
+
+        self.form_displayed = False
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user step."""
+        user_input = user_input or {}
+        errors = {}
+
+        if len(user_input) != 0:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except UnsupportedRemootioDeviceError:
+                _LOGGER.debug("Remootio device isn't supported.", exc_info=True)
+                return self.async_abort(reason="unsupported_device")
+            except vol.MultipleInvalid as e:
+                incomplete_data = True
+
+                _LOGGER.debug(
+                    f"Invalid user input. MultipleInvalid.Errors [{e.errors}]"
+                )
+                for error in e.errors:
+                    _LOGGER.debug(
+                        f"Error [{error.__class__.__name__}] Path [{error.path[0]}]"
+                    )
+                    if isinstance(error, RequiredFieldInvalid):
+                        errors[str(error.path[0])] = f"{error.path[0]}_required"
+                    else:
+                        errors[str(error.path[0])] = f"{error.path[0]}_invalid"
+                        incomplete_data = False
+
+                if incomplete_data and not self.form_displayed:
+                    errors = {}
+                    errors["base"] = "user_input_incomplete"
+            except RemootioClientConnectionEstablishmentError:
+                _LOGGER.debug("Can't connect to Remootio device.", exc_info=True)
+                errors["base"] = "cannot_connect"
+            except RemootioClientAuthenticationError:
+                _LOGGER.debug(
+                    "Can't autehnticate by the Remootio device.", exc_info=True
+                )
```

Comment:
We generally don't want exception info in logs. I would assume you already log these types on exceptions in the lib? If so there is no point in logging them here.

Reply:
Fixed.